static/panes/gccdump-view.ts

```diff
@@ -0,0 +1,423 @@
+                this.selectize.clear(true);
+                this.selectedPass = null;
+            }
+            this.updatePass(this.filters, this.selectize, result.gccDumpOutput);
+            this.showGccDumpResults(currOutput);
+
+            // enable UI on first successful compilation or after an invalid compiler selection (eg. clang)
+            if (!this.uiIsReady) {
+                this.uiIsReady = true;
+                this.onUiReady();
+            }
+        } else {
+            this.selectize.clear(true);
+            this.selectedPass = null;
+            this.updatePass(this.filters, this.selectize, false);
+            this.uiIsReady = false;
+            this.onUiNotReady();
+            if (!compiler.supportsGccDump) {
+                this.showGccDumpResults('<Tree/RTL output is not supported for this compiler (GCC only)>');
+            } else {
+                this.showGccDumpResults('<Tree/RTL output is empty>');
+            }
+        }
+        this.updateState();
+    }
+
+    getDefaultPaneName() {
+        return 'GCC Tree/RTL Viewer';
+    }
+
+    showGccDumpResults(results) {
+        this.editor.setValue(results);
+
+        if (!this.isAwaitingInitialResults) {
+            if (this.selection) {
+                this.editor.setSelection(this.selection);
+                this.editor.revealLinesInCenter(this.selection.startLineNumber, this.selection.endLineNumber);
+            }
+            this.isAwaitingInitialResults = true;
+        }
+    }
+
+    override onCompiler(compilerId: number, compiler, options: unknown, editorId: number, treeId: number) {
+        if (compilerId === this.compilerInfo.compilerId) {
+            this.compilerInfo.compilerName = compiler ? compiler.name : '';
+            this.compilerInfo.editorId = editorId;
+            this.compilerInfo.treeId = treeId;
+            this.updateTitle();
+            // TODO(jeremy-rifkin): Panes like ast-view handle the case here where the compiler doesn't support
+            // the view
+        }
+    }
+
+    override onCompilerClose(id) {
+        if (id === this.compilerInfo.compilerId) {
+            // We can't immediately close as an outer loop somewhere in GoldenLayout is iterating over
+            // the hierarchy. We can't modify while it's being iterated over.
+            this.close();
+            _.defer(function (self) {
+                self.container.close();
+            }, this);
+        }
+    }
+
+    getEffectiveFilters() {
+        return this.filters.get();
+    }
+
+    onFilterChange() {
+        this.updateState();
+        this.updateButtons();
+
+        if (this.inhibitPassSelect !== true) {
+            this.eventHub.emit('gccDumpFiltersChanged', this.compilerInfo.compilerId, this.getEffectiveFilters(), true);
+        }
+    }
+
+    override getCurrentState() {
+        const parent = super.getCurrentState();
+        const filters = this.getEffectiveFilters();
```

Comment:
Doesn't
```ts
{
  selectedPass: this.selectedPass,
  ...this.getEffectiveFilters(),
  ...parent,
}
```
work? Or is there more to it than what is seen here?

Reply:
Thank you for the catch, the only thing this will have an implication on is typing (because then ts can't prove it's a `MonacoPaneState & GccDumpState`) but I have an idea